src/v/cloud_topics/reconciler/reconciliation_consumer.cc

```diff
@@ -10,20 +10,24 @@
 
 #include "cloud_topics/reconciler/reconciliation_consumer.h"
 
-#include "model/timestamp.h"
-
 #include <seastar/core/coroutine.hh>
 
 namespace cloud_topics::reconciler {
 
 ss::future<std::optional<consumer_metadata>> build_from_reader(
   model::topic_id_partition tidp,
   model::record_batch_reader reader,
-  l1::object_builder* builder) {
+  l1::object_builder* builder,
+  reconciler_probe* probe) {
     auto gen = std::move(reader).slice_generator(model::no_timeout);
+    auto read_duration = probe->measure_l0_read_duration();
+    auto build_duration = probe->measure_object_build_duration();
+    build_duration->stop();
```

Comment:
What's the rationale for build_duration to not include start_partition(), given that's still part of object building?